pkg/cmd/roachprod/cli/chaos_disk.go

```diff
@@ -0,0 +1,160 @@
+func (cr *commandRegistry) buildChaosDiskStallCmd() *cobra.Command {
+	var nodes []int32
+	var stallType string
+	var restartNodes bool
+	var cycle bool
+	var cycleStallDuration, cycleUnstallDuration time.Duration
+
+	cmd := &cobra.Command{
+		Use:   "disk-stall <cluster>",
+		Short: "Injects disk stalls to test storage resilience",
+		Long: `Injects disk stalls to test storage resilience.
+
+Disk stall types:
+  cgroup:  Uses cgroups v2 to throttle disk I/O operations.
+           Supports partial throttling and selective stalling
+           (reads only, writes only, or including logs).
+
+  dmsetup: Uses device-mapper linear target with suspend/resume.
+           Completely freezes All disk I/O (reads and writes).
+           Requires node restart during setup to disable filesystem journaling.
+           More closely simulates a hardware disk failure.
+
+By default, stalls are maintained for 5 minutes. Use --run-forever to keep
+the stall active until manually stopped with Ctrl+C.
+
+Examples:
+  # Stall disk writes on nodes 1,2,3 using cgroup
+  roachprod chaos disk-stall mycluster --type cgroup --nodes 1,2,3
+
+  # Completely freeze disk I/O using dmsetup (simulates hardware failure)
+  roachprod chaos disk-stall mycluster --type dmsetup --nodes 1,2
+
+  # Cycle stalling on/off with custom durations
+  roachprod chaos disk-stall mycluster --type cgroup --nodes 1 \
+    --cycle --cycle-stall-duration 10s --cycle-unstall-duration 5s
+
+  # Stall until manually interrupted
+  roachprod chaos disk-stall mycluster --type cgroup --nodes 1 --run-forever
+
+  # Stall for 15 minutes before recovery
+  roachprod chaos disk-stall mycluster --type cgroup --nodes 1 --wait-before-cleanup 15m
+
+  # dmsetup stall on an insecure cluster
+  roachprod chaos disk-stall mycluster --type dmsetup --nodes 1 --wait-before-cleanup 30s --insecure
+`,
+		Args: cobra.ExactArgs(1),
+		Run: Wrap(func(cmd *cobra.Command, args []string) error {
+			clusterName := args[0]
+			nodeList := parseInt32SliceToNodes(nodes)
+
+			// Validate type
+			var failureName string
+			var failureArgs failures.DiskStallArgs
+
+			switch stallType {
+			case "cgroup":
+				failureName = failures.CgroupsDiskStallName
+				// Cgroup defaults: stall writes completely
+				// These can be exposed as flags in the future:
+				// - StallReads (default: false)
+				// - StallWrites (default: true)
+				// - StallLogs (default: false)
+				// - Throughput (default: 0 = full stall, can be 2+ for throttling)
+				failureArgs = failures.DiskStallArgs{
+					StallReads:             false,
+					StallWrites:            true,
+					StallLogs:              false,
+					Throughput:             0, // Full stall
+					RestartNodes:           restartNodes,
+					Nodes:                  nodeList,
+					Cycle:                  cycle,
+					CycleStallDuration:     cycleStallDuration,
+					CycleUnstallDuration:   cycleUnstallDuration,
+					SkipFailurePropagation: true,
```

Comment:
Hmm, why can't we just not call it from the CLI's end? It's not required for correctness from the frameworks end.